cmd/dep/testdata/harness_tests/ensure/empty/case1/final/Gopkg.toml

```diff
@@ -1,9 +1,44 @@
 
-# Example:
+## Dependencies define constraints on how dependent projects should be
+## incorporated into Gopkg.lock. They are respected by dep whether
+## this project is the current project, or if it's a dependency.
 # [[dependencies]]
+## Required: the root import path of the project being constrained
+# name = "github.com/user/project"
+## Optional: an alternate location (URL or import path) for the project's source
 # source = "https://github.com/myfork/package.git"
+## Optional, but recommended: the version constraint to enforce for the project.
+## Only one of "branch", "version" or "revision" can be specified.
+# version = "1.0.0"
 # branch = "master"
-# name = "github.com/vendor/package"
-# Note: revision will depend on your repository type, i.e git, svc, bzr etc...
+## Note: revision will depend on your repository type; git and hg have SHA1s,
+## bzr a 3-part id, svn a revision number.
 # revision = "abc123"
+
+## Overrides have the same structure as [[dependencies]], but supercede all
+## [[dependencies]] declarations from all projects. However, only the current
+## project's overrides will apply.
+##
+## Overrides are a sledgehammer, and should be used only as a last resort.
+# [[overrides]]
+## Required: the root import path of the project being constrained
+# name = "github.com/user/project"
+## Optional: an alternate location (URL or import path) for the project's source
+# source = "https://github.com/myfork/package.git"
+## Optional, but recommended: the version constraint to enforce for the project.
+## Only one of "branch", "version" or "revision" can be specified.
 # version = "1.0.0"
```

Comment:
It can be either (though note that when #225 goes in, `1.0.0` will be interpreted as `^1.0.0`). It can also specify an entirely non-semver tag for a literal match.

We need a more formal writeup of all of these rules. The examples here should really just be terse, and point to said writeup.